geo2zip/geo2zip.py

```diff
@@ -83,25 +84,64 @@ def _build_kdtree(self, data: List[Dict[str, str]]) -> Tuple[KDTree, List[str],
         tree = KDTree(coordinates)
         return tree, geoids, countries
 
-    def find_closest_zip(self, lat: float, lon: float) -> str:
+    def _haversine(self, origin_lat: float, origin_lon: float, destination_lat: float, destination_lon: float) -> float:
```

Comment:
Maybe you can use this library for haversine: https://pypi.org/project/haversine/

Reply:
You're right, let's not reinvent the wheel.